tests/acceptance/acceptancesuite.js

```diff
@@ -181,10 +208,86 @@ test('Facets, pagination, and filters do not persist accross experience links',
   await verifyCleanLink(viewAllLink);
 });
 
+fixture`Persistent Storage behavior for various components`
+  .before(setupServer)
+  .after(shutdownServer)
+  .page`${UNIVERSAL_PAGE}`;
+
+test('Navigation tab order respects persistent storage tab order', async t => {
```

Comment:
@oshi97 do we have the capacity to write Jest tests for specific components? From what I can recall, some components have an associated Jest suite, but it's clunky. I'm wondering if some of these tests can be written as Jest integration tests. That is, we create a fake instance of the component with a mocked persistent storage and try some of these things. Acceptance tests are usually meant to verify an actual user flow. They are slower to run. 

Integration tests, on the other hand, ensure that different pieces of the system work together. They are a step above unit tests. Typically, they are fairly quick to run. I have a suspicion that some of the existing acceptance tests we have may be better suited as Jest integration tests. Likely, we made them acceptance tests because it was easier to do so. If our inability to effectively mock components and their dependencies is causing us to rely too much on acceptance tests, that's an issue. In the extreme, when you have tons of acceptance tests, nobody runs them locally because they take too long. 

Verifying that the Nav component respects the tab order in persistent storage feels more like an integration test to me. There's not really a user flow that's being tested. Same thing with ensuring that referrerPageUrl is included. Conversely, I think ensuring that you stay on the same page of results after a refresh or that pagination works with the back button is an acceptance test. There's a legitimate user flow being validated. It's kinda a fine line. 

If we don't have a straightforward path for making good integration tests, that's okay. We can keep the tests I referenced acceptance tests. But, we should make an item to make it easier to create integration tests.

Reply:
I think the tests you mention could be written as jest tests, and maybe also the 'selecting a sort option updates persistent storage' and the 'the spell check component deletes skipSpellCheck'? I don't think it would be the most tedious (though probably not as straightforward as it should be). We would probably have to mock out persistent storage in the tests, which would have to be updated when we swap the storages, but we'll have to do that anyways no matter what.

Reply:
The test that make sense as acceptance tests have been moved to #1224. I will make an item for adding an integration test suite.